UIComponents/Sources/UIComponents/Components/Inputs/Content/TextFieldContainable.swift

```diff
@@ -0,0 +1,190 @@
+
+    var textColor: UIColor? { get set }
+    var font: UIFont? { get set }
+
+    var inputView: UIView? { get set }
+
+    func position(from: UITextPosition, offset: Int) -> UITextPosition?
+    func setCursorLocation(_ location: Int)
+}
+
+public final class TextFieldSingleLineView: UITextField, TextFieldContainable, UITextFieldDelegate {
+    // MARK: - Properties
+    public var shouldBeginEditing: ((TextFieldContainable) -> Bool)?
+    public var didBeginEditing: ((TextFieldContainable) -> Void)?
+    public var shouldEndEditing: ((TextFieldContainable) -> Bool)?
+    public var didEndEditing: ((TextFieldContainable) -> Void)?
+    public var shouldChangeCharacters: ((TextFieldContainable, NSRange, String) -> Bool)?
+    public var didChange: ((TextFieldContainable) -> Void)?
+    public var shouldClear: ((TextFieldContainable) -> Bool)?
+    public var shouldReturn: ((TextFieldContainable) -> Bool)?
+
+    // MARK: - Initialization
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    public required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Handling
+    @objc func textFieldDidChange(_ textField: UITextField) {
+        didChange?(self)
+    }
+
+    public func textFieldShouldBeginEditing(_ textField: UITextField) -> Bool {
+        shouldBeginEditing?(self) ?? true
+    }
+
+    public func textFieldDidBeginEditing(_ textField: UITextField) {
+        didBeginEditing?(self)
+    }
+
+    public func textFieldShouldEndEditing(_ textField: UITextField) -> Bool {
+        shouldEndEditing?(self) ?? true
+    }
+
+    public func textFieldDidEndEditing(_ textField: UITextField) {
+        didEndEditing?(self)
+    }
+
+    public func textField(
+        _ textField: UITextField,
+        shouldChangeCharactersIn range: NSRange,
+        replacementString string: String
+    ) -> Bool {
+        shouldChangeCharacters?(self, range, string) ?? true
+    }
+
+    public func textFieldShouldClear(_ textField: UITextField) -> Bool {
+        shouldClear?(self) ?? true
+    }
+
+    public func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        shouldReturn?(self) ?? true
+    }
+
+    // MARK: - Modifications
+    public func setCursorLocation(_ location: Int) {
+        guard let cursorLocation = position(from: beginningOfDocument, offset: location) else { return }
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            self.selectedTextRange = self.textRange(from: cursorLocation, to: cursorLocation)
+        }
+    }
+
+    // MARK: - Private methods
+    private func setup() {
+        delegate = self
+
+        addTarget(self, action: #selector(textFieldDidChange(_:)), for: .editingChanged)
+    }
+}
+
+public final class TextFieldMultilineContentView: UITextView, TextFieldContainable, UITextViewDelegate {
```

Comment:
In case we have `TextFieldSingleLineView`, would be better to name it in the same manner like `TextFieldMultilineView`? 🙃

Reply:
yep, good catch